src/core/mac/sub_mac.cpp

```diff
@@ -52,35 +52,48 @@ namespace Mac {
 SubMac::SubMac(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mRadioCaps(Get<Radio>().GetCaps())
-    , mState(kStateDisabled)
-    , mCsmaBackoffs(0)
-    , mTransmitRetries(0)
-    , mShortAddress(kShortAddrInvalid)
-    , mRxOnWhenBackoff(true)
-    , mEnergyScanMaxRssi(kInvalidRssiValue)
-    , mEnergyScanEndTime(0)
     , mTransmitFrame(Get<Radio>().GetTransmitBuffer())
     , mCallbacks(aInstance)
     , mPcapCallback(nullptr)
```

Comment:
Why exlcuding the `mPcapCallback` and its context from `Init()`?

Reply:
Callbacks are used by higher level software to receive events, and don't change the internal behavior of the stack (except for `mReceiveDoneCallback`). So I think resetting the stack shouldn't require the caller to set callbacks again.